### ISSUE TITLE
cli: create small bison demo

### ISSUE THREAD

**Issue by majorsylvie** [uchicago-cs/chiventure#1489]
team members: Brianna, Faruk, Kate

this task serves as part of the demo that we will present to the class, where we show bison actually doing parsing.

This can be gerryrigged in the sandbox

**Comment by majorsylvie** [comment]
Created a branch to work on this issue:
run 
```
git fetch origin
git checkout cli/1489-bison-demo
```
to use the branch!

**Comment by majorsylvie** [comment]
I believe it would be good to get bison to work with a few simple examples, such as the removal of extra prepositions like `to` and `the`.

**Comment by majorsylvie** [comment]
When coming together to work on this issue, we should define exactly what we want our demo to show in terms of examples and output.

Preliminary idea is to have it print out the list of string what would be fed to an actual existing operation.

so an example demo would be: 

normally, when the user types `"go south"`, the CLI parses it as such:
```
tokenized_cmd[0] = "GO"
tokenized_cmd[1] = "SOUTH"
```
and will raise an error if you try to give it the phrase:
`"go to the south"`

with Bison however, we have been able to give it `"go to the south"`, and it will return:
```
tokenized_cmds[0] = "GO"
tokenized_cmds[1] = "SOUTH"
```
parsing the input as all the existing chiventure architecture expects.

**Comment by bzeng6** [comment]
My idea for the Bison demo is to showcase how Bison works and its difference from the current parsing mechanism, in order to demonstrate its potential for use in future chiventure groups.

Some features to showcase:
- Taking in and outputting text
- No input word limit
- Synonyms of action key words
- Recognizing grammar for specific action kinds

**Comment by katyashchukina** [comment]
So far we have met all the aims established by Brianna above, but we have implemented each action separately. Now Brianna and I are working on generalising the commands to recognise the commands as a group, so there is only one function per kind of command rather than per each individual command. This is the main advantage of using bison so we feel this is important to implement.

**Comment by majorsylvie** [comment]
A lot of progress! we've added a readme and have a rough draft of commands we want to use to showcase bison!
We now will be working on making the output be prettier and easier to look at!
Here's the contents of the README as of 5/30/2022 7:45 pm:
```## To Compile and Run
from the working directory, (`examples/bison-demo`) run the following:
```
mkdir out
make
./parser-bison
```
This will put you into the interactive CLI interface that will communicate with and beparsed by Bison

## Cool Examples
* use my sword on the big rock
* open door
* go north
* go to the north

### Flow of Presentation:

**Opening about what we were trying to do as the new CLI team**
When trying to find a way to improve the CLI of chiventure, we thought that we couldtake the current in-house parsing and instead use a more robust parsing library.
We decided on bison, as there aren't many C libraries and of the ones we tested we were actually able to get something working.


**preamble about what bison/flex actually is and how it is relevant to chiventure.**
This would involve talking about the current implementation of parsing, along with some of the limitations.

So then, we decided that bison would then directly replace the current parsing mechanisms. This specifically involves taking the command line input and parsing it into an array of strings.
For example, when the user types "go north", this get's put into a list of stringscalled of:
[0] = "go"
[1] = "north"

so to start with, we needed to get bison to reconize the same formatting as chiventure
for examplee:

start with normal chiventure commands
eg: view stats

using bison has incredible potential for EASILY making commands more versatile andcomplex

for example, with bison is no longer a hard cap on 4 word limit per action
(WE SHOULD EITHER IMPLEMENT PARSING ON AND or just say we haven't gotten it working yet and put in backlog)

also, as you can see, the lack of a hard cap means we can support ignoringperopositions, articlees and pronouns
eg: go to the north

go north
walk north
travel north - all recognized in current grammar


# ACTUAL COMMANDS:
### OG
take sword
go north
use sword on Borja
view stats
look

## Bison is Cool!
### look at it handle prepositions and pronouns!
take the sword
go to the north
use my sword on borja
view my stats
### look at it handle synonyms!
walk north
move north
travel north

close door
shut door

take ENSU
grab ENSU

look
where

put
place


### look at it handle more interesting names!
"so instead of just going to the north, we can go to specific places!"
So I could say...:
go to Ryerson 251
take the sword of power
go to the castle of fire
use the sword of power on the evil borja```

**Comment by bzeng6** [comment]
Closing statement: 

We feel that this issue is complete because we've implemented enough grammar that we feel we can use the module during the demo and thoroughly showcase the most relevant features of Bison for Chiventure. We've included lots of documentation so that later years can use and understand our demo as well, and we've made sure that the commands on our script work.

Brianna, Kate, and Faruk all did aspects of implementation. Faruk and Brianna mainly worked on documentation.
